src/accelerate/accelerator.py

```diff
@@ -2579,7 +2581,14 @@ def save_model(
         os.makedirs(save_directory, exist_ok=True)
 
         # get the state_dict of the model
-        state_dict = self.get_state_dict(model)
+        if (
+            hasattr(model, "_hf_hook")
+            and isinstance(model._hf_hook, AlignDevicesHook)
```

Comment:
I don't think we need this since it only tells us if the model has a hook or not.

Reply:
Perfect, that simplifies the condition nicely. Removed